mod/caching/set/type_plus_right/metric/wikirate_company.rb

```diff
@@ -5,12 +5,18 @@
 end
 
 expired_cached_count_cards :set=>Type::MetricValue, :on=>[:create,:delete] do |changed_card|
-  [
-    changed_card.metric_card.fetch(:trait=>:wikirate_company),
-    # actually this is not getting cached count for `metric+company` card
-    # it is to refresh the related metric value to the company
-    # refer to ltype_rtype/metric/wikirate_company.rb
-    Card.fetch("#{changed_card.metric_name}+#{changed_card.company_name}")
+  result = [
+    # cache number of the metric values related to this metric value set
+    # for some query
+    changed_card.metric_card.fetch(trait: :wikirate_company)
   ]
+  # it is to refresh the related metric value to the company
+  # refer to ltype_rtype/metric/wikirate_company.rb
+  if (metric_card = changed_card.metric_card) &&
+     metric_card.type_id == MetricID &&
+     (company_card = changed_card.company_card) &&
+     company_card.type_id == WikirateCompanyID
+    result.push(Card.fetch("#{metric_card.name}+#{company_card.name}"))
```

Comment:
is this an appropriate place for all this random logic?  why is the code for verifying that the types are correct buried in something called expired_cached_count_cards?

I'm really concerned about WikiRate's code quality.